apache/trafficcontrol
# Workflow to check for and PR minor Go revision updates

## What does this PR (Pull Request) do?


This PR adds a GitHub Actions workflow that checks once a day if a newer minor revision exists for our Go major version (currently `1.15`). If so, the workflow will
- Create a new branch in the *apache/trafficcontrol* repo and update the `GO_VERSION` file on that branch
- Update golang.org/x/ dependencies
- Open a pull request from that branch to *apache/trafficcontrol*'s `master`  branch.

Note that the workflow will *not* create a pull request for major version updates like `1.15.7` -> `1.16.0`.


## Which Traffic Control components are affected by this PR?


- CI tests

## What is the best way to verify this PR?

1. Test the workflow
   1. Push the `go-versions-prs` branch to your own fork
   2. Go to Actions -> *Is it time to update Go?* -> *Run workflow* and manually run the workflow
   3. Verify that the pull request created in your fork contain the changes necessary to update the Go version for your fork.

2. Run the tests:
   ```shell
   cd .github/actions/pr-to-update-go
   python3 -m unittest discover ./tests
   ```

## The following criteria are ALL met by this PR
